zk.js/src/programUtxo.ts

```diff
@@ -0,0 +1,495 @@
+  const utxoHash = new BN(utxo.outUtxo.utxoHash).toArrayLike(Buffer, "be", 32);
+
+  const encryptedUtxo = await encryptOutUtxoInternal({
+    bytes,
+    utxoHash,
+    hasher,
+    account,
+    merkleTreePdaPublicKey,
+    compressed,
+    publicKey: utxo.outUtxo.publicKey,
+    encryptionPublicKey: utxo.outUtxo.encryptionPublicKey,
+  });
+  return encryptedUtxo;
+}
+
+export async function decryptProgramOutUtxo({
+  hasher,
+  encBytes,
+  account,
+  merkleTreePdaPublicKey,
+  aes,
+  utxoHash,
+  compressed = false,
+  assetLookupTable,
+  pspId,
+  pspIdl,
+  utxoName,
+}: {
+  hasher: Hasher;
+  encBytes: Uint8Array;
+  account: Account;
+  merkleTreePdaPublicKey: PublicKey;
+  aes: boolean;
+  utxoHash: Uint8Array;
+  compressed?: boolean;
+  assetLookupTable: string[];
+  pspId: PublicKey;
+  pspIdl: Idl;
+  utxoName: string;
+}): Promise<Result<ProgramOutUtxo | null, UtxoError>> {
+  const cleartext = await decryptOutUtxoInternal({
+    hasher,
+    encBytes,
+    account,
+    merkleTreePdaPublicKey,
+    aes,
+    utxoHash,
+    compressed,
+  });
+  if (!cleartext || cleartext.error || !cleartext.value) return Result.Ok(null);
+  const bytes = Buffer.from(cleartext.value);
+
+  return Result.Ok(
+    programOutUtxoFromBytes({
+      hasher,
+      bytes,
+      account,
+      assetLookupTable,
+      compressed,
+      pspId,
+      pspIdl,
+      utxoName,
+    }),
+  );
+}
+
+export type ProgramUtxo = {
+  utxo: UtxoNew;
+  pspId: PublicKey;
+  pspIdl: Idl;
+  includeUtxoData: boolean;
+  utxoData: any; // TODO: make depend on idl
+  utxoName: string;
+};
+
+export function createProgramUtxo({
+  createUtxoInputs,
+  account,
+  hasher,
+  pspId,
+  pspIdl,
+  includeUtxoData = true,
+  utxoData,
+  utxoName,
+}: {
+  createUtxoInputs: CreateUtxoInputs;
+  hasher: Hasher;
+  pspId: PublicKey;
+  pspIdl: Idl;
+  includeUtxoData?: boolean;
+  utxoData: any;
+  utxoName: string;
+  account: Account;
+}): ProgramUtxo {
+  const utxoDataInternal = utxoData;
+  checkUtxoData(utxoDataInternal, pspIdl, utxoName + "OutUtxo");
+  const utxoDataHash = createUtxoDataHash(utxoDataInternal, hasher);
+  createUtxoInputs["utxoDataHash"] = utxoDataHash.toString();
+  createUtxoInputs["verifierAddress"] = pspId;
+
+  const utxo = createUtxo(hasher, account, createUtxoInputs, false);
+  const programOutUtxo: ProgramUtxo = {
+    utxo,
+    pspId,
+    pspIdl,
+    includeUtxoData,
+    utxoData: utxoDataInternal,
+    utxoName,
+  };
+  return programOutUtxo;
+}
+
+export async function decryptProgramUtxo(
+  encBytes: Uint8Array,
+  account: Account,
+  merkleTreePdaPublicKey: PublicKey,
+  aes: boolean,
+  utxoHash: Uint8Array,
+  hasher: Hasher,
+  compressed: boolean = true,
+  merkleProof: string[],
+  merkleTreeLeafIndex: number,
+  assetLookupTable: string[],
+  pspId: PublicKey,
+  pspIdl: Idl,
+  utxoName: string,
+): Promise<Result<ProgramUtxo | null, UtxoError>> {
+  const decryptedProgramOutUtxo = await decryptProgramOutUtxo({
```

Comment:
given the large amount of parameters i'd suggest we use object destructuring syntax

Reply:
i.e. ({ encBytes: ... , account: ..} : { encBytes: Uint8Array; account: Account,...})